cmd/qos.go

```diff
@@ -58,10 +64,16 @@ func getServiceQoSInstances(
 			if _, ok := hydratorServiceIDsIdx[serviceID]; ok {
 				hydratorQoSGenerators[serviceID] = evmEndpointStore
 			}
+
+		// TODO_FUTURE: The logic here is complex enough to justify using a builder/factory function pattern here.
+		// At-least having something like func buildSolanaQoSInstance(...) in a solana.go file either here or under
+		// config package will make the initialization/configuration code easier to read and maintain.
 		case config.ServiceIDSolana:
```

Comment:
@adshmh Isn't this already implemented?

Reply:
Not yet, the initialization logic is getting more complex here with the additional conditions and services, and needs to be split into several logical, cohesive units.

Reply:
@adshmh Do you mind if I add a TODO and tag you in it (as I'm not clear on what this entails)?

Reply:
Do it